WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/viewmodels/AnalogWatchFaceViewModel.kt

```diff
@@ -15,123 +15,52 @@
-
-    // Launches a new coroutine to insert the data in a non-blocking way
-    private fun updateAnalogWatchFace(analogWatchFaceEntity: AnalogWatchFaceEntity) = scope.launch {
-        repository.updateAnalogWatchFace(analogWatchFaceEntity)
-    }
-
-    // [WatchFaceColorStyleEntity] operations:
-    suspend fun getAllWatchFaceColorStyles() =
-        repository.getAllWatchFaceColorStyles()
-
-    // [WatchFaceArmDimensionsEntity] operations:
-    private suspend fun getWatchFaceArmDimensions(watchFaceArmDimensionsId: String) =
-        repository.getWatchFaceArmDimensions(watchFaceArmDimensionsId)
-
-    // Launches a new coroutine to insert the data in a non-blocking way
-    private fun updateWatchFaceArmDimensions(watchFaceArmDimensionsEntity: WatchFaceArmDimensionsEntity) = scope.launch {
-        repository.updateWatchFaceArmDimensions(watchFaceArmDimensionsEntity)
-    }
-
-    // Updates multiple table entries if the user changes something in the settings.
-    fun updateUserStylesInDatabase(analogWatchFaceId: Int, userStyle: UserStyle) = scope.launch {
-        // Returns if a valid [AnalogWatchFaceEntity] isn't returned for the id.
-        val analogWatchFaceEntity = getAnalogWatchFace(analogWatchFaceId) ?: return@launch
-
-        updateColorStyleAndHourPips(userStyle, analogWatchFaceEntity)
-        updateMinuteArmLength(userStyle, analogWatchFaceEntity.minuteHandDimensionsId)
-    }
-
-    // Updates the color style and hour hand pips (ticks on the outside of the watch) in the
-    // associated [AnalogWatchFaceEntity] in the database.
-    private suspend fun updateColorStyleAndHourPips(
-        userStyle: UserStyle,
-        analogWatchFaceEntity: AnalogWatchFaceEntity
-    ) {
-        var revisedAnalogWatchFaceEntity = analogWatchFaceEntity
-
-        // 1. Updates color style associated with the watch face.
-        val newColorStyleSetting: String =
-            userStyle.toMap()[COLOR_STYLE_SETTING]
-                ?: revisedAnalogWatchFaceEntity.activeColorStyleId
+    private val _uiState = MutableStateFlow(UserChangesUiState.Success(WatchFaceUserChanges()))
+    // The UI collects from this StateFlow to get its state updates
+    val uiState: StateFlow<UserChangesUiState> = _uiState.asStateFlow()
 
-        if (revisedAnalogWatchFaceEntity.activeColorStyleId != newColorStyleSetting) {
-            revisedAnalogWatchFaceEntity =
-                revisedAnalogWatchFaceEntity.copy(activeColorStyleId = newColorStyleSetting)
-        }
-
-        // 2. Updates Hour Pips (dashes around the outside of the watch).
-        val newHoursPipsSettingStringVersion: String =
-            userStyle.toMap()[DRAW_HOUR_PIPS_STYLE_SETTING] ?: ""
-
-        val newHoursPipsSetting =
-            if (newHoursPipsSettingStringVersion.isEmpty()) {
-                revisedAnalogWatchFaceEntity.drawHourPips
-            } else {
-                newHoursPipsSettingStringVersion.toBoolean()
+    init {
+        scope.launch {
```

Comment:
This would be a great place for a more declarative version with `stateIn`:

```kotlin
    val uiState: StateFlow<UserChangesUiState> =
        watchFaceUserSettingChangesDataSource.getUserWatchFaceChanges()
            .map(UserChangesUiState::Success)
            .stateIn(
                scope = scope,
                started = SharingStarted.Eagerly,
                initialValue = UserChangesUiState.Success(WatchFaceUserChanges())
            )
```

`SharingStarted.Eagerly` would be the equivalent replacement, but `SharingStarted.WhileSubscribed` might be good as well to avoid doing any unnecessary work if nobody is listening.